backend/src/models/tag.model.ts

```diff
@@ -1,19 +1,18 @@
+import { AggregateRoot } from '@nestjs/cqrs';
 import { prop } from '@typegoose/typegoose';
 import { Expose } from 'class-transformer';
-import { Item } from './item.model';
 
-export class Tag {
+export class Tag extends AggregateRoot {
   @Expose()
   @prop({ required: true })
   name: string;
 
-  // TODO: Add relevant Account data
-
   @Expose()
-  @prop({ required: true })
-  slug: string;
+  @prop({ required: true }) // Default is 1, because when you create it, it is attached to an Item
+  count?: number;
```

Comment:
Prop is required true but still optional here?

Reply:
Yep didn't remove that my bad